algorithms/number_theory/eratosthenes_sieve.py

```diff
@@ -0,0 +1,59 @@
+"""Module with the implementation of eratosthenes sieve."""
+
+
+from typing import List
+
+
+class EratosthenesSieve:
+    """Class implementation of the eratosthenes sieve algorithm."""
+
+    def __init__(self, n: int):
+        self._n = n
+        self._no_cousins = [False for _ in range(self._n + 1)]
```

Comment:
Can we rename all the `cousins` to `primes`? I think it is a confusing name that comes from the literal translation of Spanish. The `prime` in English comes from being `primary` or `primarios` in Spanish, not `cousins` or `primos`. 😉

Reference: https://primes.utm.edu/notes/faq/WhyCalledPrimes.html

Also, why should we have a list called `_no_primes` when we can have `_primes` instead? I think it is always easier to avoid using negation in our names. It avoids confusion, I would say.

Consider changing the name and the logic.

Reply:
ok